src/main/host/process.rs

```diff
@@ -948,6 +954,40 @@ impl Process {
         self.check(host);
     }
 
+    /// Adds a new thread to the process and schedules it to run.
+    /// Intended for use by `clone`.
+    pub fn add_thread(&self, host: &Host, thread: ThreadRef) {
+        let pid = self.id();
+        let tid = thread.id();
+        let thread = RootedRc::new(host.root(), RootedRefCell::new(host.root(), thread));
+        self.threads.borrow_mut().insert(tid, thread);
+
+        // Schedule thread to start. We're giving the caller's reference to thread
+        // to the TaskRef here, which is why we don't increment its ref count to
+        // create the TaskRef, but do decrement it on cleanup.
+        let task = TaskRef::new(move |host| {
+            let (cprocess, cthread) = {
+                let Some(process) = host.process_borrow(pid) else {
+                    // This might happen if a thread calls `clone` and then `exit_group`
+                    debug!("Process {:?} no longer exists. Can't start its thread {}.", pid, tid);
+                    return;
+                };
+                let process = process.borrow(host.root());
+                let threads = process.threads.borrow();
+                let Some(thread) = threads.get(&tid) else {
+                    // Maybe possible e.g. if a thread is targeted with `tgkill` before it
+                    // gets a chance to start.
+                    debug!("Thread {} no longer exists. Can't start it.", tid);
```

Comment:
Could you just move the `RootedRc<RootedRefCell<ThreadRef>>` into the `TaskRef` closure and use it directly instead of looking up the thread? Then you wouldn't need to worry about it not existing?

Reply:
I think we still wouldn't want to do anything with it here in that case. We could store a `WeakRootedRc` instead of the `ThreadId` to save the lookup, but I like having the `ThreadId` for the log message in the case that the object no longer exists.